### PR TITLE
LPD-43256 CommercePriceList and CommerceOrder - Ability to use the Currency's ERC instead of the Currency Code

### PR DESCRIPTION
**Pull request by luca-pellizzon**

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce-core-api/src/main/java/com/liferay/headless/commerce/core/util/CurrencyUtil.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+				return commerceCurrency;
+			}
+		}
+
+		if (Validator.isNull(currencyExternalReferenceCode)) {
+			commerceCurrency =
+				CommerceCurrencyLocalServiceUtil.
+					fetchCommerceCurrencyByExternalReferenceCode(
+						currencyExternalReferenceCode, companyId);
+
```
Comment (brianikim) [inline modules/apps/commerce/headless/headless-commerce-core-api/src/main/java/com/liferay/headless/commerce/core/util/CurrencyUtil.java:50]: @luca-pellizzon , Is this and line 25 intentional? Wouldn't `commerceCurrency` always be null with these checks? We are passing in null `currencyCode` and null `currencyExternalReferenceCode`

Comment (luca-pellizzon) [reply modules/apps/commerce/headless/headless-commerce-core-api/src/main/java/com/liferay/headless/commerce/core/util/CurrencyUtil.java:50]: must be a isNotNull, thanks :+1:

Comment (brianikim) [reply modules/apps/commerce/headless/headless-commerce-core-api/src/main/java/com/liferay/headless/commerce/core/util/CurrencyUtil.java:50]: Besides this, everything else looks good to me.

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:forward:force

**Pull request closed without merging**